lang/java/reef-runtime-yarn/src/main/java/org/apache/reef/runtime/yarn/driver/restart/DFSEvaluatorLogAppendReaderWriter.java

```diff
@@ -65,6 +66,11 @@ public synchronized void writeToEvaluatorLog(final String formattedEntry) throws
     }
   }
 
+  @Override
+  public Iterable<String> readFromEvaluatorLog() throws IOException {
```

Comment:
I don't think it's a good idea to read the entire file upon `Iterator` creation, since the file could be large. Right now, the design opens the file at the creation of the `Iterator` and closes the file when there are no more lines to read (when `hasNext()` returns `false`).